aslam_backend/src/LinearSystemSolver.cpp

```diff
@@ -65,19 +68,19 @@ namespace aslam {
           indices[i + 1] = indices[i] + nJPerThread;
         // deal with the remainder.
         indices.back() = _errorTerms.size();
-        // Build a thread pool and evaluate the jacobians.
-        boost::thread_group threads;
-        std::vector<SafeJob> jobs(nThreads);
-        for (size_t i = 0; i < nThreads; ++i) {
-          jobs[i] = SafeJob(boost::bind(job, i, indices[i], indices[i + 1], useMEstimator));
-          threads.create_thread(boost::ref(jobs[i]));
+
+        //replace by std::async for performance
```

Comment:
This comment does not really make sense to stay because later the reader doesn't know what is replaced. 
```suggestion
```